src/core/reducers/autocomplete.js

```diff
@@ -0,0 +1,69 @@
+export const AUTOCOMPLETE_LOADED = 'core/autocomplete/AUTOCOMPLETE_LOADED';
+export const AUTOCOMPLETE_STARTED = 'core/autocomplete/AUTOCOMPLETE_STARTED';
+export const AUTOCOMPLETE_CANCELLED = 'core/autocomplete/AUTOCOMPLETE_CANCELLED';
+
+const initialState = {
+  loading: false,
+  suggestions: [],
+};
+
+export function autocompleteStart({ errorHandlerId, filters }) {
+  if (!errorHandlerId) {
+    throw new Error('errorHandlerId is required');
+  }
+
+  if (!filters) {
+    throw new Error('filters are required');
+  }
+
+  return {
+    type: AUTOCOMPLETE_STARTED,
+    payload: { errorHandlerId, filters },
+  };
+}
+
+export function autocompleteCancel() {
+  return { type: AUTOCOMPLETE_CANCELLED };
+}
+
+export function autocompleteLoad({ results }) {
+  if (!results) {
+    throw new Error('results are required');
+  }
+
+  return {
+    type: AUTOCOMPLETE_LOADED,
+    payload: { results },
+  };
+}
+
+export default function reducer(state = initialState, action = {}) {
+  const { payload } = action;
+
+  switch (action.type) {
+    case AUTOCOMPLETE_CANCELLED:
+      return {
+        ...state,
+        loading: false,
+        suggestions: [],
+      };
+    case AUTOCOMPLETE_STARTED:
+      return {
```

Comment:
To avoid some repetition and guarantee that the state gets reset, this could be done as:

```js
return { ...initialState, loading: true };
```